types/common-tags/index.d.ts

```diff
@@ -1,15 +1,17 @@
-// Type definitions for common-tags v1.4.0
+// Type definitions for common-tags v1.7.2
 // Project: https://github.com/declandewet/common-tags
 // Definitions by: Viktor Zozuliak <https://github.com/zuzusik>
 //                 Paul Wang <https://github.com/tzupengwang>
+//                 coolreader18 <https://github.com/coolreader18>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 declare module 'common-tags' {
-    type TemplateTag = (literals: TemplateStringsArray, ...placeholders: any[]) => string;
+    type TemplateTag = (literals: TemplateStringsArray, ...placeholders: any[]) => any;
```

Comment:
I just wonder - what is the case for `any`? Shouldn't it always return the string? Or the library supports anything and then converts it to string?

If there are some special cases - we should specify types more precisely. Using `any` should be very exceptional.

Reply:
I replied by email previously, so you might not have gotten a notification:

My use case, for example, is a regex tag; if there are any substitutions
that are an instanceof regexp it replaces them with the source and in the
end it turns it all into a regex. common-tagd doesn't type check any of it,
so if you were to switch to typescript you might otherwise have type errors
with it.